vermouth/processors/water_bias.py

```diff
@@ -0,0 +1,151 @@
+        """
+        self.water_bias = water_bias
+        self.auto_bias = auto_bias
+        self.idr_regions = idr_regions
+        self.system = None
+
+    def assign_residue_water_bias(self, molecule, res_graph):
+        """
+        Assign the residue water bias for all residues
+        with a secondary structure element or that are
+        defined by the region selector. Region selectors
+        supercede the auto assignment.
+
+        Parameters
+        ----------
+        molecule: :class:`vermouth.molecule.Molecule`
+            the molecule
+        res_graph: :class:`vermouth.molecule.Molecule`
+            the residue graph of the molecule
+        """
+        for res_node in res_graph.nodes:
+            resid = res_graph.nodes[res_node]['resid']
+            _old_resid = res_graph.nodes[res_node]['_old_resid']
+            chain = res_graph.nodes[res_node]['chain']
+            resname = res_graph.nodes[res_node]['resname']
+
+            if _in_resid_region(_old_resid, self.idr_regions):
+                eps = self.water_bias.get('idr', 0.0)
+            elif self.auto_bias:
+                sec_struc = res_graph.nodes[res_node]['cgsecstruct']
+                eps = self.water_bias.get(sec_struc, 0.0)
+            else:
+                continue
+
+            if eps == 0.0:
+                continue
+
+            vs_go_node = next(get_go_type_from_attributes(res_graph.nodes[res_node]['graph'],
+                                                          resid=resid,
+                                                          chain=chain,
+                                                          prefix=molecule.meta.get('moltype')))
+
+            # what is the blocks bb-type
+            bb_type = molecule.force_field.blocks[resname].nodes['BB']['atype']
+            size = _get_bead_size(bb_type)
+            # bead sizes are defined in the force-field file as
+            # regular, small and tiny
+            sigma = float(molecule.force_field.variables[size])
+            # update interaction parameters
+            atoms = (molecule.force_field.variables['water_type'], vs_go_node)
+            water_bias = NonbondParam(atoms=atoms,
+                                      sigma=sigma,
+                                      epsilon=eps,
+                                      meta={"comment": ["water bias", sec_struc]})
+            self.system.gmx_topology_params["nonbond_params"].append(water_bias)
+
+    def run_molecule(self, molecule):
+        """
+        Assign water bias for a single molecule
+        """
+        if not self.system:
+            raise IOError('This processor requires a system.')
+
+        if not molecule.meta.get('moltype'):
+            raise ValueError('The molecule does not have a moltype name.')
+
+        if hasattr(molecule, 'res_graph'):
+            res_graph = molecule.res_graph
+        else:
+            res_graph = make_residue_graph(molecule)
```

Comment:
I would rather not. In principle the proper way would be to generate that graph once before and not hope another processor secretly creates it. So if it is there we use it otherwise we don't secretly propagate.

Reply:
Alright. 
I really like the idea of having Molecules have their resgraph as attribute (or cached property). The problem is that Molecule objects cannot keep track of when or for how long this residue graph is valid, since that can change based on atom properties (and atoms are just dicts). 
Option 1) Atoms get an Atom (Bead, Particle) class, which communicates with their parent molecule, and we all get a migraine
Option 2) Gentleman's agreement between Processors, that if you do something which may cause cached residue boundaries to change that you then invalidate the cache. When a res_graph is requested, the cache gets populated again. Relatively easy, but I promise you it'll cause a bug down the line because a Processor forgets to invalidate the cache.

Either way, not this PR.